tests/virtctl/usbredir.go

```diff
@@ -187,77 +174,108 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 })
 
 func runConnectGoroutine(
-	virtClient kubecli.KubevirtClient,
 	name string,
 	namespace string,
 	ctx context.Context,
 	errch chan error,
 ) {
-	defer GinkgoRecover()
-
-	usbredirStream, err := virtClient.VirtualMachineInstance(namespace).USBRedir(name)
-	if err != nil {
-		errch <- err
-		return
-	}
-	usbredirConnect(usbredirStream, ctx)
-}
-
-func usbredirConnect(
-	stream kvcorev1.StreamInterface,
-	ctx context.Context,
-) {
-
-	usbredirClient, err := usbredir.NewUSBRedirClient(ctx, "localhost:0", stream)
-	Expect(err).ToNot(HaveOccurred())
-
-	usbredirClient.ClientConnect = func(inCtx context.Context, device, address string) error {
+	cmd := newVirtctlCommand("usbredir",
+		"--namespace", namespace,
+		"--no-launch",
+		name,
+	)
+	// To find ip/port to connect
+	rOut, wOut := io.Pipe()
+	cmd.SetOut(wOut)
+	defer rOut.Close()
+
+	// To find errors
+	rErr, wErr := io.Pipe()
+	cmd.SetErr(wErr)
+	defer rErr.Close()
+
+	remote := make(chan error)
+	go func() {
 		defer GinkgoRecover()
-		conn, err := net.Dial("tcp", address)
-		Expect(err).ToNot(HaveOccurred())
-		defer conn.Close()
-
-		buf := make([]byte, 1024, 1024)
-
-		// write hello message to remote (VMI)
-		if nw, err := conn.Write([]byte(helloMessageLocal)); err != nil {
-			return err
-		} else {
-			Expect(nw).To(Equal(len(helloMessageLocal)))
-		}
-
-		// reading hello message from remote (VMI)
-		if nr, err := conn.Read(buf); err != nil {
-			return err
-		} else {
-			Expect(buf[0:nr]).ToNot(BeEmpty(), "response should not be empty")
-			Expect(buf[0:nr]).To(HaveLen(len(helloMessageRemote)))
-		}
+		remote <- cmd.Execute()
+	}()
 
-		// Signal connected after read/write to be sure no TCP operation failed too
-		if connected, ok := inCtx.Value("connected").(chan struct{}); ok {
-			connected <- struct{}{}
+	go func(r *io.PipeReader) {
+		defer GinkgoRecover()
+		scanner := bufio.NewScanner(r)
+		if scanner.Scan() {
+			// stderr should only be logging errors but we catch only known ones
+			line := scanner.Text()
+			if !strings.Contains(line, "websocket: bad handshake") {
```

Comment:
Shouldn't this not be negated?

Reply:
It is actually wrong. My mistake when moving to use `newVirtctlCommand`.
The idea was the opposite, if it isn't `bad handshake` error, we should return `unexpected` one. I'll fix and test locally shortly.

Reply:
Removed.